cypress/components/search/search.cy.tsx

```diff
@@ -328,73 +313,74 @@ context("Test for Search component", () => {
+      const callback: SearchProps["onClick"] = cy.stub().as("onClick");
+
       CypressMountWithProviders(
         <Search onClick={callback} defaultValue={testCypress} searchButton />
       );
 
-      searchButton()
-        .click()
-        .then(() => {
-          // eslint-disable-next-line no-unused-expressions
-          expect(callback).to.have.been.calledOnce;
-        });
+      searchButton().click();
+      cy.get("@onClick").should("have.been.calledOnce");
+      
     });
 
     it("should call onChange callback when a type event is triggered", () => {
+      const callback: SearchProps["onChange"] = cy.stub().as("onChange");
+
       CypressMountWithProviders(<SearchComponent onChange={callback} />);
 
-      searchDefaultInput()
-        .type("1")
-        .then(() => {
-          // eslint-disable-next-line no-unused-expressions
-          expect(callback).to.have.been.calledOnce;
-        });
+      searchDefaultInput().type("1");
+      cy.get("@onChange").should("have.been.calledOnce");
+      
     });
 
     it("should call onFocus callback when a focus event is triggered", () => {
+      const callback: SearchProps["onFocus"] = cy.stub().as("onFocus");
+
       CypressMountWithProviders(<SearchComponent onFocus={callback} />);
 
-      searchDefaultInput()
-        .focus()
-        .then(() => {
-          // eslint-disable-next-line no-unused-expressions
-          expect(callback).to.have.been.calledOnce;
-        });
+      searchDefaultInput().focus();
+      cy.get("@onFocus").should("have.been.calledOnce");  
     });
 
     it("should call onBlur callback when a blur event is triggered", () => {
+      const callback: SearchProps["onBlur"] = cy.stub().as("onBlur");
+
       CypressMountWithProviders(<SearchComponent onBlur={callback} />);
 
-      searchDefaultInput()
-        .focus()
-        .blur()
-        .then(() => {
-          // eslint-disable-next-line no-unused-expressions
-          expect(callback).to.have.been.calledOnce;
-        });
+      searchDefaultInput().focus().blur();
+      cy.get("@onBlur").should("have.been.calledOnce");    
     });
 
-    it.each([["Enter"], ["Space"]])(
+    it.each([keysToTrigger[0], keysToTrigger[1]])(
       "should call onKeyDown callback when a keyboard event is triggered",
       (key) => {
+        const callback: SearchProps["onKeyDown"] = cy.stub().as("onKeyDown");
+
         CypressMountWithProviders(<SearchComponent onKeyDown={callback} />);
 
-        searchDefaultInput()
-          .trigger("keydown", keyCode(key))
-          .then(() => {
-            // eslint-disable-next-line no-unused-expressions
-            expect(callback).to.have.been.calledOnce;
-          });
+        searchDefaultInput().trigger("keydown", keyCode(key));
+        cy.get("@onKeyDown").should("have.been.calledOnce"); 
+          
       }
     );
+
+    it("should have the expected border radius styling when no search button enabled", () => {
```

Comment:
I would move this and the next one test out from the `events` describe and move up to the check props for Search component